### PR TITLE
Simplify SurveyRepository tests using in memory local db instead of mocks

### PR DESCRIPTION
**Pull request by gino-m**
Also pushes launch of `activateSurvey()` up to view models so that exceptions can be propagated and handled there.

Towards #1471.

@shobhitagarwal1612 Could you PTAL?

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt [hunk 1]
```diff
@@ -109,15 +108,9 @@ constructor(
       return
     }
 
-    externalScope.launch {
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt:112]: Please cleanup the injected dependency as well.

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt [hunk 2]
```diff
@@ -17,106 +17,88 @@ package com.google.android.ground.repository
...
   @Inject lateinit var localValueStore: LocalValueStore
-
   @DefaultDispatcher @Inject lateinit var testDispatcher: CoroutineDispatcher
 
   @Before
   override fun setUp() {
     super.setUp()
-    `when`(mockLocalSurveyStore.insertOrUpdateSurvey(any())).thenReturn(Completable.complete())
+    // Reset local test db between tests.
+    localDatabase.clearAllTables()
```
Comment (shobhitagarwal1612) [inline ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt:56]: Isn't this implicit?

**Review by shobhitagarwal1612 — CHANGES_REQUESTED** [review]

File: ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt (hunk 2 above)
Comment (gino-m) [reply ground/src/test/java/com/google/android/ground/repository/SurveyRepositoryTest.kt]: I suppose it is. Each test gets a different instance of `LocalDatabase`, which I assume has a different SQLite behind it. Thanks for pointing this out.

**Review by shobhitagarwal1612 — APPROVED** [review]

**Review by shobhitagarwal1612 — APPROVED** [review]
LGTM!

**Pull request merged**